x-pack/examples/gen_ai_streaming_response_example/server/plugin.ts

```diff
@@ -0,0 +1,114 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { CreateChatCompletionResponse } from 'openai';
+import { Readable } from 'stream';
+import { Plugin, CoreSetup } from '@kbn/core/server';
+import { schema } from '@kbn/config-schema';
+import { PluginStartContract as ActionsPluginStart } from '@kbn/actions-plugin/server';
+
+interface GenAiStreamingResponseExamplePluginStart {
+  actions: ActionsPluginStart;
+}
+
+interface Message {
+  role: string;
+  content: string;
+}
+
+interface MessageBody {
+  model?: string;
+  messages: Message[];
+}
+
+export class GenAiStreamingResponseExamplePlugin implements Plugin<void, void> {
+  public setup({ http, getStartServices }: CoreSetup<GenAiStreamingResponseExamplePluginStart>) {
+    const router = http.createRouter();
+
+    router.get(
+      {
+        path: `/internal/examples/get_gen_ai_connectors`,
+        validate: {},
+      },
+      async (_, request, response) => {
+        const [, { actions }] = await getStartServices();
+        const actionsClient = await actions.getActionsClientWithRequest(request);
+
+        const allConnectors = await actionsClient.getAll();
+
+        return response.ok({
+          body: (allConnectors ?? []).filter(
+            (connector) => connector.actionTypeId === '.gen-ai' && !connector.isPreconfigured
+          ),
+        });
+      }
+    );
+
+    router.post(
+      {
+        path: `/internal/examples/execute_gen_ai_connector`,
+        validate: {
+          body: schema.object({
+            connector_id: schema.string(),
+            prompt: schema.string(),
+          }),
+        },
+      },
+      async (_, request, response) => {
+        const [, { actions }] = await getStartServices();
+        const actionsClient = await actions.getActionsClientWithRequest(request);
+
+        const connector = await actionsClient.get({ id: request.body.connector_id });
+
+        let messageBody: MessageBody;
+        if (connector.config?.apiProvider === 'OpenAI') {
+          messageBody = {
+            model: 'gpt-3.5-turbo',
+            messages: [
+              {
+                role: 'user',
+                content: request.body.prompt,
+              },
+            ],
+          };
+        } else if (connector.config?.apiProvider === 'Azure OpenAI') {
+          messageBody = {
+            messages: [
+              {
+                role: 'user',
+                content: request.body.prompt,
+              },
+            ],
+          };
+        } else {
+          throw Boom.badRequest(
+            `Invalid generative AI connector selected - ${connector.config?.apiProvider} is not a valid provider`
+          );
+        }
+
+        return response.ok({
+          body: (
+            await actionsClient.execute({
+              actionId: request.body.connector_id,
+              params: {
+                subAction: 'test',
```

Comment:
I'm not sure if I follow why we need to call `test` here? What does `test` do vs `run`?

Reply:
OK! just to be sure: if I want a streaming response, I should use test, and there are no other implications other than that it looks weird? I've also got a followup question: what happens when we start calling more endpoints, or add more LLMs? Do we all handle that in this connector?